### ISSUE TITLE
Halt rendering on error?

### ISSUE THREAD

**Issue by xbc5** [xbc5/hexo-csp#20]
Should this be done? It seems like not halting the build on a security violation would be bad for CI.

Find a way to do this properly.

Issues:
1. the `after_render` filter fires for each path, throwing within that means throwing hundreds of errors;
2. throwing within the root index means throwing for the CLI too, since it also loads plugins -- it makes no sense to do this;
3. the hexo testing utils hates exceptions, and causes unhandled rejection error;

In short: can't throw on load, during rendering, or during testing.
